src/charm.py

```diff
@@ -96,12 +95,54 @@ def _are_relations_ready(self) -> bool:
         return self._tls.get_tls_relation() is not None
 
     def _reconcile(self) -> None:
-        """Reconcile charm status based on configuration and integrations."""
+        """Reconcile charm status based on configuration and integrations.
+
+        Raises:
+            RuntimeError: when initializing the lightkube client fails,
+            or when creating the gateway resource fails.
+        """
+        try:
+            # Set field_manager for server-side apply when patching resources
+            # Keep this consistent across client initializations
+            kubeconfig = KubeConfig.from_service_account()
+            client = Client(config=kubeconfig, field_manager=self.app.name)
+        except ConfigError as exc:
+            LOGGER.error("Error initializing the lightkube client: %s", exc)
+            raise RuntimeError("Error initializing the lightkube client.") from exc
+
+        try:
+            gateway_resource_definition = GatewayResourceDefinition.from_charm(self)
+        except InvalidCharmConfigError as exc:
+            LOGGER.error("Invalid charm config: %s", exc.msg)
+            self.unit.status = BlockedStatus("Invalid charm configuration")
+            return
+
         tls_certificates_relation = self._tls.get_tls_relation()
         if not tls_certificates_relation:
             self.unit.status = BlockedStatus("Waiting for TLS.")
             return
-        self.unit.status = ActiveStatus()
+
+        gateway_resource_manager = GatewayResourceManager(
+            namespace=gateway_resource_definition.namespace,
+            labels=self._labels,
+            client=client,
+        )
+
+        try:
+            gateway = gateway_resource_manager.define_resource(gateway_resource_definition)
+        except (CreateGatewayError, InvalidResourceError) as exc:
+            LOGGER.error("Error creating the gateway resource %s", exc)
+            raise RuntimeError("Cannot create gateway.") from exc
+        except KuberentesCreateResourceError as exc:
```

Comment:
s/Kuberentes/Kubernetes

What the user should do in this case once the charm is blocked?

Reply:
I believe exc.msg will tell the user to do juju trust --scope=cluster.